install_files/ansible-base/roles/ossec/tasks/register.yml

```diff
@@ -0,0 +1,99 @@
+---
+- name: Check whether Application Server is registered as OSSEC agent.
+  command: /var/ossec/bin/list_agents -a
+  register: ossec_list_agents_result
+  # Read-only task, so don't report changed
+  when: ossec_is_server
+  changed_when: false
+
+# Gnarly vars retrieval logic in this task. The "register" action above applies
+# only to the Monitor Server, so the Application Server won't be able to access
+# the value (and the play will fail as a result). So on both hosts, let's look up
+# the registered value by referencing the hostvars for the Monitor Server, then
+# copy the result to a more conveniently named fact on both hosts.
+- name: Set host fact for OSSEC registration state.
+  set_fact:
+    ossec_agent_already_registered: "{{ hostvars[groups.securedrop_monitor_server.0].ossec_list_agents_result.stdout == app_hostname +'-'+app_ip+' is available.' }}"
```

Comment:
so i ran into a similar issue that i had before ... `hostvars[groups.securedrop_monitor_server.0]` doesnt populate at all for me. Running under tails `3.3`, ansible `2.3.2.0`, i added a debug right before the `set_fact` here ...

`debug: var="{{ hostvars[groups.securedrop_monitor_server.0] }}"

response is `VARIABLE IS NOT DEFINED` on both servers and then  `start authd` is always firing for me. I had to fix a commit for the iptables rules --- and now its working... but its always firing. Note that I am running this with explicit virtualenv, and the under the ansible dir:

`./securedrop-prod.xml -t ossec --diff --ask-become-pass`